src/arc_utilities/ros_helpers.py

```diff
@@ -85,3 +83,20 @@ def _get_subscribers(topic_path: str):
         if sub[0] == topic_path:
             subs.extend(sub[1])
     return subs
+
+
+def try_to_connect(*publishers, raise_on_fail: bool = False):
+    connected = [p.get_num_connections() > 0 for p in publishers]
```

Comment:
nit: If you accept the other changes, you could move this inside the `for` loop and remove the duplicated ln 95